render/widget.go

```diff
@@ -29,3 +29,47 @@ func ModInt(a, m int) int {
 	}
 	return a
 }
+
```

Comment:
The following functions are quite useful to "de-clutter" code. Otherwise there is tons of casts to `float64` and back to `int`, which affect readability.